### PR TITLE
Feature/subgraph-init

### PR DESCRIPTION
**Pull request by Rekard0**
-draft ERD
-draft mapping
-draft template

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,6 @@ yarn-error.log*
 .env.test.local
 .env.production.local
 
+# generated
+generated
+subgraph.yaml
```
Comment (nivida) [inline .gitignore]: EOF

File: .prettierrc [hunk 2]
```diff
@@ -1,5 +1,19 @@
 {
-	"bracketSpacing": false,
-	"singleQuote": true,
-	"arrowParens": "avoid"
+  "bracketSpacing": false,
```
Comment (nivida) [inline .prettierrc:2]: Yeah, I was meaning in the chat that you better keep this change and the code style changes in general out of this PR. But we can keep it for now and discuss the prettier configuration later on together.

File: packages/subgraph/manifest/data/rinkeby.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "info": "# Do not edit subgraph.yaml,this is a generated file. \n# Instead, edit subgraph.placeholder.yaml and run: yarn manifest-[network-name]",
+  "network": "rinkeby",
```
Comment (nivida) [inline packages/subgraph/manifest/data/rinkeby.json]: Rinkeby is not the correct network :))

File: packages/subgraph/manifest/subgraph.placeholder.yaml [hunk 4]
```diff
@@ -0,0 +1,175 @@
+{{info}}
+specVersion: 0.0.2
+description: A template for Aragon subgraphs
+repository: https://github.com/zaragoza
```
Comment (nivida) [inline packages/subgraph/manifest/subgraph.placeholder.yaml]: ```aragon/zaragoza```

File: packages/subgraph/manifest/subgraph.placeholder.yaml [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+      kind: ethereum/events
+      apiVersion: 0.0.5
+      language: wasm/assemblyscript
+      file: ./src/process.ts
+      entities:
+        - Execution
+        - Process
+      abis:
+        - name: StoppableProcess
+          file: ./../contracts/artifacts/contracts/core/processes/stoppable/StoppableProcess.sol/StoppableProcess.json
```
Comment (nivida) [inline packages/subgraph/manifest/subgraph.placeholder.yaml:125]: Paths in general should use the constants as discussed :)

File: packages/subgraph/manifest/subgraph.placeholder.yaml [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        - event: CastVote(indexed uint256,indexed address,bool,uint256)
+          handler: handleCastVote
+        - event: ChangeMinQuorum(uint64)
+          handler: handleChangeMinQuorum
+        - event: ChangeSupportRequired(uint64)
+          handler: handleChangeSupportRequired
+        - event: ExecuteVote(indexed uint256)
+          handler: handleExecuteVote
+        - event: StartVote(indexed uint256,indexed address,string)
+          handler: handleStartVote
```
Comment (nivida) [inline packages/subgraph/manifest/subgraph.placeholder.yaml]: EOF

File: packages/subgraph/package.json [hunk 7]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "@aragon/zaragoza-subgraph",
+  "version": "0.0.1",
+  "description": "Aragon V2 Subgraph",
+  "homepage": "https://github.com/aragon/zaragoza#readme",
```
Comment (nivida) [inline packages/subgraph/package.json]: this could point directly to the subgraph package :)

File: packages/subgraph/package.json [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  "license": "MIT",
+  "scripts": {
```
Comment (nivida) [inline packages/subgraph/package.json:7]: The whole graph setup should be in a docker container.

File: packages/subgraph/package.json [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    "codegen": "graph codegen",
+    "build": "graph build",
+    "deploy": "graph deploy --node https://api.studio.thegraph.com/deploy/ sarkawttest",
+    "create-local": "graph create --node http://localhost:8020/ sarkawttest",
+    "remove-local": "graph remove --node http://localhost:8020/ sarkawttest",
+    "deploy-local": "graph deploy --node http://localhost:8020/ --ipfs http://localhost:5001 sarkawttest",
+    "test": "rm -r tests/.bin; graph test",
+    "coverage": "graph test -c",
+    "manifest-rinkeby": "scripts/build-manifest.sh rinkeby"
```
Comment (nivida) [inline packages/subgraph/package.json]: We better should test directly with Arbitrum.

File: packages/subgraph/schema.graphql [hunk 10]
```diff
@@ -0,0 +1,123 @@
+# Enums
+enum ExecutionState {
+  RUNNING
+  STOPPED
+  HALTED
+  EXECUTED
+}
+
+enum VoterState {
```
Comment (nivida) [inline packages/subgraph/schema.graphql]: Related this we have to discuss of how we handle such things in a more generic way. Especially if we have an out look into the future of this product.

File: packages/subgraph/schema.graphql [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+  allowedAction: Bytes!
+  process: Process!
+  isRemoved: Boolean!
+}
+
+type Process @entity {
+  id: ID!
+  dao: Dao!
+  address: Bytes!
+  isActive: Boolean!
```
Comment (nivida) [inline packages/subgraph/schema.graphql]: isActive?

File: packages/subgraph/schema.graphql [hunk 12]
```diff
@@ -0,0 +1,123 @@
...
+  metadata: Bytes!
+  additionalArguments: Bytes
+  actions: [Action!]! @derivedFrom(field: "proposal")
+}
+
+type AllowedAction @entity {
+  id: ID!
+  allowedAction: Bytes!
+  process: Process!
+  isRemoved: Boolean!
```
Comment (nivida) [inline packages/subgraph/schema.graphql]: isRemoved?

File: packages/subgraph/schema.graphql [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+type Process @entity {
+  id: ID!
+  dao: Dao!
```
Comment (nivida) [inline packages/subgraph/schema.graphql]: What if a process is used by multiple DAOs in the future? eg a sub-dao?

File: packages/subgraph/schema.graphql [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+  sender: Bytes!
+  amount: BigInt!
+  reason: String!
+}
+
+# Daos
+type Dao @entity {
+  id: ID!
+  name: String!
+  daoAddress: Bytes!
```
Comment (nivida) [inline packages/subgraph/schema.graphql]: can't we use the address as ID?

File: packages/subgraph/scripts/build-manifest.sh [hunk 15]
```diff
@@ -0,0 +1,13 @@
...
+FILE=$NETWORK'.json'
+DATA=manifest/data/$FILE
+
+echo 'Generating manifest from data file: '$DATA
+cat $DATA
+
+mustache \
+  $DATA \
+  manifest/subgraph.placeholder.yaml \
+  > subgraph.yaml
```
Comment (nivida) [inline packages/subgraph/scripts/build-manifest.sh]: EOF

File: packages/subgraph/src/dao.ts [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+  }
+}
+
+export function handleExecuted(event: Executed): void {
+  // TODO:
+}
+
+export function handleDeposited(event: Deposited): void {
+  let daoId = event.address.toHexString();
+  let id =
```
Comment (nivida) [inline packages/subgraph/src/dao.ts:74]: If we have such IDs concatenated based on event data should we I think create a keccak hash out of it to save some stored bytes.

File: packages/subgraph/src/process.ts [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+    actionEntity.value = action.value;
+    actionEntity.data = action.data;
+
+    actionEntity.proposal = id;
+
+    actionEntity.save();
+  }
+
+  // handle proposal
+  let proposalEntiry = new Proposal(id);
```
Comment (nivida) [inline packages/subgraph/src/process.ts]: typo😇

**Review by nivida — CHANGES_REQUESTED** [review]
Added a few comments.. but definitely have to think about the data model closer asap.

File: packages/subgraph/manifest/data/rinkeby.json (hunk 3 above)
Comment (Rekard0) [reply packages/subgraph/manifest/data/rinkeby.json]: yes, sure, each network and testnet will have its own JSON down the road

File: .gitignore (hunk 1 above)
Comment (Rekard0) [reply .gitignore]: done

File: packages/subgraph/manifest/subgraph.placeholder.yaml (hunk 4 above)
Comment (Rekard0) [reply packages/subgraph/manifest/subgraph.placeholder.yaml]: done

File: packages/subgraph/manifest/subgraph.placeholder.yaml (hunk 6 above)
Comment (Rekard0) [reply packages/subgraph/manifest/subgraph.placeholder.yaml]: done

File: packages/subgraph/package.json (hunk 7 above)
Comment (Rekard0) [reply packages/subgraph/package.json]: done

File: .prettierrc (hunk 2 above)
Comment (Rekard0) [reply .prettierrc:2]: opps, ok

File: packages/subgraph/schema.graphql (hunk 10 above)
Comment (Rekard0) [reply packages/subgraph/schema.graphql]: yah, forgot to remove

File: packages/subgraph/scripts/build-manifest.sh (hunk 15 above)
Comment (Rekard0) [reply packages/subgraph/scripts/build-manifest.sh]: done

File: packages/subgraph/src/process.ts (hunk 17 above)
Comment (Rekard0) [reply packages/subgraph/src/process.ts]: thanks, done

File: packages/subgraph/src/dao.ts (hunk 16 above)
Comment (Rekard0) [reply packages/subgraph/src/dao.ts:74]: well, yes we can, but i noticed this pattern of concatenating is a common practice, it is easier for reading while debugging...etc
i would rather do that during the task of testing against a testnet
but sure we can think of optimizations like this.

File: packages/subgraph/schema.graphql (hunk 14 above)
Comment (Rekard0) [reply packages/subgraph/schema.graphql]: yes, thanks

File: packages/subgraph/package.json (hunk 8 above)
Comment (Rekard0) [reply packages/subgraph/package.json:7]: i made another task for this,,, should it be include it in this PR?

File: packages/subgraph/manifest/subgraph.placeholder.yaml (hunk 5 above)
Comment (Rekard0) [reply packages/subgraph/manifest/subgraph.placeholder.yaml:125]: i made another task for this,,, should it be include it in this PR?

File: packages/subgraph/schema.graphql (hunk 13 above)
Comment (Rekard0) [reply packages/subgraph/schema.graphql]: thanks, we should use many-to-many relationship
done

File: packages/subgraph/package.json (hunk 9 above)
Comment (Rekard0) [reply packages/subgraph/package.json]: i made another task for this,,, should it be include it in this PR?

File: packages/subgraph/manifest/subgraph.placeholder.yaml (hunk 5 above)
Comment (nivida) [reply packages/subgraph/manifest/subgraph.placeholder.yaml:125]: Feel free to do it in a separate PR to not blow this one up even more

File: packages/subgraph/package.json (hunk 9 above)
Comment (nivida) [reply packages/subgraph/package.json]: 👍

File: packages/subgraph/src/dao.ts (hunk 16 above)
Comment (nivida) [reply packages/subgraph/src/dao.ts:74]: Hmm good point.. cause I was in this situation once.. and it annoyed me cause I was not able to check things in a few seconds.. 👍

File: packages/subgraph/package.json (hunk 8 above)
Comment (nivida) [reply packages/subgraph/package.json:7]: 👍

**Review by nivida — APPROVED** [review]
Approved! :) 

The mentioned required improvements will be done as discussed in other smaller PRs to not blow this one up even more.

**Pull request merged**